java-type-checker/java_type_checker/expressions.py

```diff
@@ -66,6 +84,21 @@ def __init__(self, lhs, rhs):
         self.lhs = lhs
         self.rhs = rhs
 
+    def static_type(self):
+        return self.lhs.static_type()
+    
+    def check_types(self):
+        self.lhs.check_types()
+        self.rhs.check_types()
+        
+        if self.lhs.static_type() == self.rhs.static_type() or self.rhs.static_type().is_subtype_of(self.lhs.static_type()):
```

Comment:
Your version of this method is a bit more concise and easier to understand, but they both work and do the same thing.